### PR TITLE
add validation for input definition fields

### PR DESCRIPTION
**Pull request by linhvo**
## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: input_definition.go [hunk 1]
```diff
@@ -246,3 +271,28 @@ func (i *InputDefinition) AddFrame(frame InputFrame) error {
 	}
 	return nil
 }
+
```
Comment (raskle) [inline input_definition.go]: A valid action needs a Frame name and it must be one from the set defined in the frames definition.

**Comment by linhvo** [comment]
@raskle: I updated nil RowID

File: input_definition.go [hunk 2]
```diff
@@ -89,16 +99,36 @@ func (i *InputDefinition) LoadDefinition(pb *internal.InputDefinition) error {
...
 
+	numPrimaryKey := 0
+	countRowID := make(map[string]uint64)
 	for _, field := range pb.Fields {
 		var actions []Action
 		for _, action := range field.Actions {
+			if err := i.ValidateAction(action); err != nil {
+				return err
+			}
+			if action.RowID != 0 && action.Frame != "" {
```
Comment (raskle) [inline input_definition.go]: Instead of checking for rowID !=0
if action.ValueDestination == "single-row-boolean"  && action.Frame != "" {

Comment (linhvo) [inline input_definition.go]: oh I missed that, good point.

**Pull request merged**